src/IssueViz.Security/Hotspots/ServerHotspotStore.cs

```diff
@@ -0,0 +1,104 @@
+﻿/*
+ * SonarLint for Visual Studio
+ * Copyright (C) 2016-2023 SonarSource SA
+ * mailto:info AT sonarsource DOT com
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using SonarQube.Client;
+using SonarQube.Client.Models;
+
+namespace SonarLint.VisualStudio.IssueVisualization.Security.Hotspots
+{
+    internal interface IServerHotspotStore
+    {
+        Task UpdateAsync(string projectKey, string branch, CancellationToken token);
+
+        Task<IList<SonarQubeHotspotSearch>> GetServerHotspotsAsync(string projectKey, string branch, CancellationToken token);
+
+        event EventHandler<ServerHotspotStoreUpdatedEventArgs> ServerHotspotStoreUpdated;
+    }
+
+    [Export(typeof(IServerHotspotStore))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    internal class ServerHotspotStore : IServerHotspotStore
+    {
+        private readonly ISonarQubeService sonarQubeService;
+        internal /*for testing*/ readonly static Dictionary<string, IList<SonarQubeHotspotSearch>> serverHotspots = new Dictionary<string, IList<SonarQubeHotspotSearch>>();
+
+        public event EventHandler<ServerHotspotStoreUpdatedEventArgs> ServerHotspotStoreUpdated;
+
+        [ImportingConstructor]
+        public ServerHotspotStore(ISonarQubeService sonarQubeService)
+        {
+            this.sonarQubeService = sonarQubeService;
+        }
+
+        public async Task UpdateAsync(string projectKey, string branch, CancellationToken token)
```

Comment:
This method keeps info for all compositekeys. While it could be useful, we need to assess that versus the likelihood of this being unused (aka just a memory leak)